packages/core-data/src/fetch/deprecated.js

```diff
@@ -0,0 +1,17 @@
+/**
+ * WordPress dependencies
+ */
+import deprecated from '@wordpress/deprecated';
+
+/**
+ * Internal dependencies
+ */
+import fetchLinkSuggestions from './fetch-link-suggestions';
+
+export const __experimentalFetchLinkSuggestions = ( search ) => {
+	deprecated( 'wp.coreData.__experimentalFetchLinkSuggestions', {
+		since: '6.1',
+		alternative: 'wp.coreData.fetchLinkSuggestions',
+	} );
```

Comment:
```suggestion
	deprecated( 'wp.coreData.__experimentalFetchLinkSuggestions', {
		since: '6.1',
		version: '6.2',
		alternative: 'wp.coreData.fetchLinkSuggestions',
	} );
```

Thanks for proposing this alternative and for your well reasoned arguments about why it might be necessary.
 
Please could I check - are you suggesting that we support this prefixed version indefinitely? 

I understand the arguments around soft deprecation and backwards compatibility in WordPress. However, it does set a precedent that any future experimental APIs should also be handled thus and therefore will remain in the codebase forever.

The contract was always "these are experimental and might be changed without warning". That is undermined _slightly_ by how long the API has been experimental, but the expectations were set out in the project.

Perhaps a good middle ground would be that we will _**remove**_ this code in `6.2` ([docs](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-deprecated/#default)). That addresses both:

- the concern around bloating the codebase.
- the concern about hard deprecating APIs that have been around for a long time.

It affords folks time to update their codebases whilst also retaining the spirit of the `__experimental` contract.

What do you think @youknowriad @mtias?

Reply:
Yes, I am proposing this be maintained well in to the future.

I disagree that was the contract in WordPress Core, it was only the contract in the plugin.

> This strategy won’t be applied in Core because it’s very important that we continue to maintain the user trust with auto-updates and **there’s no intention in changing that in the future**.
_Original emphasis_, [source](https://make.wordpress.org/core/2018/11/14/technical-organisation-post-5-0/)

As I mentioned on #40316, this lose of faith in auto-updates is happening.

We, WordPress Committers, made a mistake exposing the experimental APIs in WordPress Core. I deliberately use the word "we" because that includes me for not monitoring editor commits as closely as I could have prior to my role on the release squad for WP 6.0.